webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,256 @@
+            defaultMessage='Moving this thread changes who has access'
+            values={{
+                strong: (x: React.ReactNode) => <strong>{x}</strong>,
+            }}
+        />
+    );
+
+    const notification = (
+        <NotificationBox
+            variant={'info'}
+            text={notificationText}
+            id={'move_thread'}
+        />
+    );
+
+    const handlePostError = (error: ClientError) => {
+        setIsButtonClicked(false);
+        setPostError(error.message);
+        setHasError(true);
+        setTimeout(() => setHasError(false), Constants.ANIMATION_TIMEOUT);
+    };
+
+    const handleSubmit = async () => {
+        setIsButtonClicked(true);
+        if (!selectedChannel) {
+            setIsButtonClicked(false);
+            return;
+        }
+
+        const channel = selectedChannel.details;
+
+        let result = await actions.moveThread(post.root_id || post.id, channel.id);
+
+        if (result.error) {
+            handlePostError(result.error);
+            return;
+        }
+
+        if (
+            channel.type === Constants.MENTION_MORE_CHANNELS &&
+            channel.type === Constants.OPEN_CHANNEL
+        ) {
+            result = await actions.joinChannelById(channel.id);
+
+            if (result.error) {
+                handlePostError(result.error);
+                return;
+            }
+        }
+
+        result = await actions.switchToChannel(channel);
+
+        if (result.error) {
+            handlePostError(result.error);
+            return;
+        }
+
+        onHide();
+    };
+
+    const postPreviewFooterMessage = formatMessage({
+        id: 'move_thread_modal.preview.footer_message',
+        defaultMessage: 'Originally posted in ~{channelName}',
+
+    },
+    {
+        channelName: originalChannel.display_name,
+    });
+
+    return (
+        <GenericModal
+            className='a11y__modal forward-post move-thread'
+            id='move-thread-modal'
+            show={true}
+            autoCloseOnConfirmButton={false}
+            compassDesign={true}
+            modalHeaderText={formatMessage({
+                id: 'move_thread_modal.title',
+                defaultMessage: 'Move thread',
+            })}
+            confirmButtonText={formatMessage({
+                id: 'move_thread_modal.button.forward',
+                defaultMessage: 'Move',
+            })}
+            cancelButtonText={formatMessage({
+                id: 'move_thread_modal.button.cancel',
+                defaultMessage: 'Cancel',
+            })}
+            isConfirmDisabled={isButtonClicked}
+            handleConfirm={handleSubmit}
+            handleEnterKeyPress={handleSubmit}
+            handleCancel={onHide}
+            onExited={onHide}
+        >
+            <div
+                className={'move-thread__body'}
+                ref={measuredRef}
+            >
+                {notification}
+                <ForwardPostChannelSelect
```

Comment:
The name of the component is misleading, since now it does more than it used to do. Maybe not in the scope right now, but if you do the refactor I also commented before (moving this to its own component) it would also need a new name for the more generalized functionality.